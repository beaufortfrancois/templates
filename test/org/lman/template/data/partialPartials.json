{
  "a": "outer a",
  "b": "outer b",
  "c": {
    "a": "inner a",
    "b": "inner b"
  }
}
