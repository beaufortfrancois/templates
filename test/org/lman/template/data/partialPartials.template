This tests partials that are passed an incomplete set of arguments bound to
local variables.

Rendering in outer context:
-> bound to the same name
{{+partials.one moo:(partials.two a:a b:b) /}}
-> bound to swapped names
{{+partials.one moo:(partials.two a:b b:a) /}}

{{#c}}
Rendering in inner context with pre-bound variables:
-> bound to the same name
{{+partials.one moo:(partials.two a:a b:b) /}}
-> bound to swapped names
{{+partials.one moo:(partials.two a:b b:a) /}}
{{/c}}

{{#c}}
Rendering in inner context with post-bound variables:
-> bound to the same name
{{+partials.two a:a b:b d:hello /}}
-> bound to swapped names
{{+partials.two a:b b:a d:hello /}}
{{/c}}

Done.
