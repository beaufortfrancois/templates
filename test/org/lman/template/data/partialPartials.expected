This tests partials that are passed an incomplete set of arguments bound to
local variables.

Rendering in outer context:
-> bound to the same name
I am rendering partial (a=outer a, b=outer b) now.
-> bound to swapped names
I am rendering partial (a=outer b, b=outer a) now.

Rendering in inner context with pre-bound variables:
-> bound to the same name
I am rendering partial (a=inner a, b=inner b) now.
-> bound to swapped names
I am rendering partial (a=inner b, b=inner a) now.

Rendering in inner context with post-bound variables:
-> bound to the same name
(a=inner a, b=inner b)
-> bound to swapped names
(a=inner b, b=inner a)

Done.
